### DISCUSSION TITLE
When to use Postgres?

### DISCUSSION THREAD

**Discussion by jecorn** [mealie-recipes/mealie#2306] [Q&A] [answered]
I'm really loving mealie v1.0b, but have been noticing slow load times with a database of ca. 6,000 recipes (2/3 with images). I'm wondering if migrating to a Postgres database would be worth trying.

I'm seeing that infinite scroll with background access works well once pages are loaded. But moving from the front page to All Recipes takes several seconds. But I'm not really dealing with a concurrency use case. There will only be one person using mealie at a time. Is this a potential Postgres use case?

I would go ahead and just try it, but I ran into some issues when setting up a Postgres container parallel to the sqlite one (e.g. going to Recipe Data in Data Management with such a large database seems to cause the whole instance to hang and never shows recipes to export) and so wanted to check with the experts before I go ahead with troubleshooting.

**Comment by michael-genson** [comment]
There are some major performance improvements in nightly, with 6k recipes I recommend trying it out if you're not already

Sounds like a good opportunity to use Postgres too, but I'm using sqlite so I'm no help there

**Reply by jecorn** [reply]
Thanks for the tip. I’ll start with nightly. Any db migration needed, or just pull the new image?

**Reply by michael-genson** [reply]
If you're already on 1.0.0b just pulling the new images should work fine. I'd grab a backup though

**Reply by jecorn** [reply]
Performance is incredibly improved with SQLite on nightly. Great job, devs!

**Comment by hay-kot** [comment, marked as answer]
To speak generally about 

> When to use Postgres?

Is mostly about concurrent usage. If you have 10+ users, it's probably a good idea to migrate from SQLite because of concurrent access to the database. SQLite is extremely fast and for 99% of users of Mealie is probably as fast as Postgres. Most of the slowness is from bad SQL queries that need to be improved, which thanks to some awesome contributors has been improved GREATLY in the last few months.
